src/new/sendTransaction.ts

```diff
@@ -132,11 +131,12 @@ class SendTransaction extends EventEmitter {
       tokens: [],
     };
     const chooseInputs = this.inputs.length === 0;
-    const tokenSet = new Set<string>();
+    // Map of token uid to the chooseInputs value of this token
+    const tokenMap = new Map<string, boolean>();
 
     for (const output of this.outputs) {
       if (isDataOutput(output)) {
-        tokenSet.add(HTR_UID);
+        tokenMap.set(HTR_UID, false);
```

Comment:
Why don't you set this as `true` just like the other?

Reply:
Done